pydantic_settings/main.py

```diff
@@ -572,7 +572,7 @@ class CliAppBaseSettings(BaseSettings, model_cls):  # type: ignore
 
             model = CliAppBaseSettings(**model_init_data)
             model_init_data = {}
-            for field_name, field_info in model.model_fields.items():
+            for field_name, field_info in model.__pydantic_fields__.items():
```

Comment:
```suggestion
            for field_name, field_info in type(model).model_fields.items():
```

Assuming `model` is an instance